src/builds.ts

```diff
@@ -0,0 +1,162 @@
+   * Manifest](https://github.com/GoogleChrome/http2-push-manifest) for your
+   * application.
+   */
+  addPushManifest?: boolean;
+
+  /**
+   * A config file that's passed to the [sw-precache
+   * library](https://github.com/GoogleChrome/sw-precache). See [its
+   * README](https://github.com/GoogleChrome/sw-precache) for details of the
+   * format of this file.
+   *
+   * Ignored if `addServiceWorker` is not `true`.
+   *
+   * Defaults to `"sw-precache-config.js`.
+   */
+  swPrecacheConfig?: string;
+
+  /**
+   * Insert prefetch link elements into your fragments so that all dependencies
+   * are prefetched immediately. Add dependency prefetching by inserting `<link
+   * rel="prefetch">` tags into entrypoint and `<link rel="import">` tags into
+   * fragments and shell for all dependencies.
+   */
+  insertPrefetchLinks?: boolean;
+
+  /**
+   * By default, fragments are unbundled. This is optimal for HTTP/2-compatible
+   * servers and clients.
+   *
+   * If the --bundle flag is supplied, all fragments are bundled together to
+   * reduce the number of file requests. This is optimal for sending to clients
+   * or serving from servers that are not HTTP/2 compatible.
+   */
+  bundle?: boolean;
+
+  /** Options for processing HTML. */
+  html?: {
+    /** Minify HTMl by removing comments and whitespace. */
+    minify?: boolean
+  };
+
+  /** Options for processing CSS. */
+  css?: {
+    /** Minify inlined and external CSS. */
+    minify?: boolean
+  };
+
+  /** Options for processing JavaScript. */
+  js?: {
+    /** Minify inlined and external JavaScript. */
+    minify?: boolean,
+
+    /** Use babel to compile all ES6 JS down to ES5 for older browsers. */
+    compile?: boolean
+  };
+}
+
+export const buildPresets = new Map<string, ProjectBuildOptions>([
+  ['es5-bundled', {
+    name: 'es5-bundled',
+    js: {minify: true, compile: true},
+    css: {minify: true},
+    html: {minify: true},
+    bundle: true,
+    addServiceWorker: true,
+    addPushManifest: true,
+    insertPrefetchLinks: true,
+  }],
+  ['es6-bundled', {
+    name: 'es6-bundled',
+    js: {minify: true, compile: false},
+    css: {minify: true},
+    html: {minify: true},
+    bundle: true,
+    addServiceWorker: true,
+    addPushManifest: true,
+    insertPrefetchLinks: true,
+  }],
+  ['es6-unbundled', {
+    name: 'es6-unbundled',
+    js: {minify: true, compile: false},
+    css: {minify: true},
+    html: {minify: true},
+    bundle: false,
+    addServiceWorker: true,
+    addPushManifest: true,
+    insertPrefetchLinks: true,
+  }],
+]);
+
+export function isValidPreset(presetName: string) {
+  return buildPresets.has(presetName);
+}
+
+/**
+ * Apply a build preset (if a valid one exists on the config object) by
+ * deep merging the given config with the preset values.
+ */
+export function applyBuildPreset(config: ProjectBuildOptions) {
+  const presetName = config.preset;
+  if (!presetName || !isValidPreset(presetName)) {
```

Comment:
This is currently handled in the `validate()` stage. This separation of parsing and validating was what prompted this discussion: https://github.com/Polymer/polymer-project-config/issues/18

Okay to merge this way? I think the long-term solution here is to handle validation during parsing (and not as a separate step) but I'd rather handle that refactoring in a future PR>